dali/pipeline/data/tensor_vector.cc

```diff
@@ -395,18 +389,21 @@ std::shared_ptr<TensorList<Backend>> TensorVector<Backend>::AsTensorList(bool ch
 
 template <typename Backend>
 void TensorVector<Backend>::resize_tensors(int new_size) {
-  auto old_size = tensors_.size();
-  tensors_.resize(new_size);
-  for (int i = old_size; i < new_size; i++) {
-    tensors_[i] = std::move(std::make_shared<Tensor<Backend>>());
-    tensors_[i]->set_pinned(pinned_);
+  if (new_size > tensors_.size()) {
+    auto old_size = curr_tensors_size_;
+    tensors_.resize(new_size);
+    for (int i = old_size; i < new_size; i++) {
+      tensors_[i] = std::move(std::make_shared<Tensor<Backend>>());
+      tensors_[i]->set_pinned(is_pinned());
+    }
   }
+  curr_tensors_size_ = new_size;
```

Comment:
```C++
} else if (new_size < curr_tensors_size_) {
  for (int i = new_size; i < curr_tensors_size_; i++) {
    if (tensors_[i]->shares_data())
      tensors_[i]->Reset();
  }
}
```
this is critical for:
- maintaining correct `views_count_`
- properly releasing reference to data shared with external object.

Reply:
Done